File: 2645. Minimum Additions to Make Valid String/Solution.java
class Solution {
    public int addMinimum(String word) {
        int n = word.length(), res = 0, curr = 0;
        char[] ch = word.toCharArray();
        for (char c : ch) {
            while (c - 'a' != curr) {
                res++;
                curr = (curr + 1) % 3;
            }
            curr = (curr + 1) % 3;
        }
        while (curr != 0) {
            res++;
            curr = (curr + 1) % 3;
        }
        return res;
    }
}

File: 2645. Minimum Additions to Make Valid String/Solution.py
class Solution:
    def addMinimum(self, word: str) -> int:
        res, curr = 0, 0
        for c in word:
            while ord(c) - ord('a') != curr:
                res += 1
                curr = (curr + 1) % 3
            curr = (curr + 1) % 3
        while curr != 0:
            res += 1
            curr = (curr + 1) % 3
        return res